prdoc/pr_5623.prdoc

```diff
@@ -0,0 +1,32 @@
+# Schema: Polkadot SDK PRDoc Schema (prdoc) v1.0.0
+# See doc at https://raw.githubusercontent.com/paritytech/polkadot-sdk/master/prdoc/schema_user.json
+
+title: Generic slashing side-effects
+
+doc:
+  - audience: Runtime Dev
+    description: |
+      What?
+      Make it possible for other pallets to implement their own logic when a slash on a balance occurs.
+
+      Why?
+      In the introduction of holds @gavofyork said:
+
+      Since Holds are designed to be infallibly slashed, this means that any logic using a Freeze must handle the possibility of the frozen amount being reduced, potentially to zero. A permissionless function should be provided in order to allow bookkeeping to be updated in this instance.
+
+      At Polimec we needed to find a way to reduce the vesting schedules of our users after a slash was made, and after talking to @kianenigma at the Web3Summit, we realized there was no easy way to implement this with the current traits, so we came up with this solution.
```

Comment:
I don't think all this context is needed in the PRDoc. This will go into the changelog. The changelog will already be quite big so I'd include only the thing the audience needs to know

Reply:
agreed. PRDoc should be the "outcome", not the thought process.